packages/chakra-ui/src/Skeleton/index.js

```diff
@@ -0,0 +1,69 @@
+/** @jsx jsx */
+import { Fragment } from "react";
+import { useTheme } from "../ThemeProvider";
+import { useColorMode } from "../ColorModeProvider";
+import { css, jsx, keyframes } from "@emotion/core";
+import Box from "../Box";
+
+const skeletonGlow = (colorStart, colorEnd) => keyframes`
+from {
+  border-color: ${colorStart};
+  background: ${colorStart};
+}
+
+to {
+  border-color: ${colorEnd};
+  background: ${colorEnd};
+}
+`;
+
+const getStyle = ({ colorStart, colorEnd }) => css`
+  border-color: ${colorStart} !important;
+  box-shadow: none !important;
+
+  // do not !important this for Firefox support
+  background: ${colorStart};
+
+  // Prevent background color from extending to the border and overlappping
+  background-clip: padding-box !important;
+  cursor: default;
+
+  // Transparent text will occupy space but be invisible to the user
+  color: transparent !important;
+  animation: 1s linear infinite alternate ${skeletonGlow(colorStart, colorEnd)};
+  pointer-events: none;
+  user-select: none;
+
+  // Make pseudo-elements (CSS icons) and children invisible
+  &::before,
+  &::after,
+  * {
+    visibility: hidden !important;
+  }
+`;
+
+const Skeleton = props => {
+  const { colors } = useTheme();
+  const { colorMode } = useColorMode();
+  const defaultStart = { light: colors.gray[100], dark: colors.gray[800] };
+  const defaultEnd = { light: colors.gray[400], dark: colors.gray[500] };
+  const {
+    colorStart = defaultStart[colorMode],
+    colorEnd = defaultEnd[colorMode],
+    isLoaded = false,
+    ...rest
+  } = props;
+  if (isLoaded) {
+    return <Fragment {...props} />;
+  }
```

Comment:
This looks amazing! Thank you for working on it. Here are my thoughts:

1. In real-life scenarios, we might not need the `isLoaded` prop since rendering a Skeleton will be based on some data-fetching logic and it might look like this:

```jsx
function Card(){
    const { data, loading, error } = useFetcher("/api/chakra")
    if(isLoading) return <Skeleton />
    if (error) return <ErrorDisplay /> 
    return <Component />
}
```

Hence, I'd suggest we remove the `isLoaded` prop. I hope this makes sense?

2. Expose ability to change the animation duration,  we could call this prop `speed ` whose default value is 1,  and pass that to the CSS template string in  Line 33. Make sense?

Other than that, it's good to go for the first version. In the future, we might consider creating primitives like `Circle`, `Square`, `Picture` loaders just like `react-content-loader` so it's even easier to use.

Thanks again.

Reply:
I added the `isLoaded` prop for cases like this
```tsx
function Card() {
    const { data, loading, error } = useFetcher('/api/chakra')
    if (error) return <ErrorDisplay />
    return (
        <Box>
            <Skeleton isLoaded={!loading}>
                <Heading>{data?.title}</Heading>
            </Skeleton>
        </Box>
    )
}
```
Where i want to show the skeleton in a child component,
if there was not `isLoaded` prop i would need to  do this
```tsx
function Card() {
    const { data, loading, error } = useFetcher('/api/chakra')
    if (error) return <ErrorDisplay />
    return (
        <Box>
            {isLoading ? (
                <Skeleton>
                    <Heading> </Heading>
                </Skeleton>
            ) : (
                <Heading>{data?.title}</Heading>
            )}
        </Box>
    )
}
```
I prefer the first one, it is simpler and i use the child component to give the same size to the skeleton

I will add the speed prop

Reply:
Hmm. I agree with you. In this case, we might not want to forward all props to the `Fragment`, we should only pass children to it, to prevent any React warnings.

```jsx
  if (isLoaded) {
    return <Fragment> {props.children} </Fragment>;
  }
```